### DISCUSSION TITLE
Motion masks - do they conserve CPU?

### DISCUSSION THREAD

**Discussion by jbrizz** [blakeblackshear/frigate#7005] [Ask A Question] [answered]
I'd had a look around but can't find this specific question clarified. [This](https://github.com/blakeblackshear/frigate/issues/3860) mentions it but it isn't clear.

I understand motion masks prevent object detection from beginning within the masked area, but if an object is detected outside a masked area and moves into the masked area it is still tracked.

My question is, if I mask off areas such as trees and bushes that move around on windy days, is this area then effectively not monitored for motion (unless a tracked object moves into it), thus conserving CPU, or still monitored for motion but just unable to begin tracking an object?

Thanks!

**Comment by NickM-27** [comment, marked as answer]
Motion masks mean that area always has the same pixel values (because motion is just comparing the current values to the values in the current frame), essentially meaning there can never be motion in that area because the values never change. 

So it does save some CPU in that the motion masked areas won't be sent for object detection

**Reply by jbrizz** [reply]
So probably valuable for optimising CPU usage then. Thanks.
